adr/0003-jsonpath-support.md

```diff
@@ -0,0 +1,43 @@
+# 3. JSONPath support
+
+Date: 15-03-2024
+
+## Status
+
+Draft
+
+## Context
+
+Support for JsonPATH was requested by some potential library clients.  
+However, JsonPATH RFC specifies way too many features which are tedious to implement without having runtime dependencies.  
+Therefore, we have to decide which features are necessary and which are not. 
+
+## Decisions
+
+The decisions are premised on the following expectations:
+1. Most of the clients will not use JsonPATH keys.  
+2. The correctness of the library takes precedent over the amount of features.
```

Comment:
correctness, performance and maintainability always come before any non-requested feature basically